vim-engine/src/main/kotlin/com/maddyhome/idea/vim/action/motion/text/MotionWordEndAction.kt

```diff
@@ -21,41 +21,53 @@ import com.maddyhome.idea.vim.handler.Motion
 import com.maddyhome.idea.vim.handler.Motion.AbsoluteOffset
 import com.maddyhome.idea.vim.handler.MotionActionHandler
 
+// Vim considers an empty line as a word/WORD, but for vi compatibility, `e` and `E` do not stop at empty lines (`ge`,
+// `gE` and `w`/`W` and `b`/`B` do)
 @CommandOrMotion(keys = ["gE"], modes = [Mode.NORMAL, Mode.VISUAL, Mode.OP_PENDING])
-class MotionBigWordEndLeftAction : WordEndAction(Direction.BACKWARDS, true)
+class MotionBigWordEndLeftAction : WordEndAction(Direction.BACKWARDS, bigWord = true, stopAtEmptyLine = true)
 
 @CommandOrMotion(keys = ["E"], modes = [Mode.NORMAL, Mode.VISUAL, Mode.OP_PENDING])
-class MotionBigWordEndRightAction : WordEndAction(Direction.FORWARDS, true)
+class MotionBigWordEndRightAction : WordEndAction(Direction.FORWARDS, bigWord = true, stopAtEmptyLine = false)
 
 @CommandOrMotion(keys = ["ge"], modes = [Mode.NORMAL, Mode.VISUAL, Mode.OP_PENDING])
-class MotionWordEndLeftAction : WordEndAction(Direction.BACKWARDS, false)
+class MotionWordEndLeftAction : WordEndAction(Direction.BACKWARDS, bigWord = false, stopAtEmptyLine = false)
 
```

Comment:
`stopAtEmptyLine = true`?

Reply:
Good spot! As it happens, this flag is not used for backwards ("left") motion, so it's not relevant. I've updated the comment and added `true` as the default parameter so this is less surprising.